Dockerfile

```diff
@@ -7,9 +7,18 @@ COPY go.mod go.sum ./
 RUN go mod download
 COPY . .
 RUN go build -ldflags="-X 'github.com/kanopy-platform/buildah-plugin/internal/version.version=${VERSION}' -X 'github.com/kanopy-platform/buildah-plugin/internal/version.gitCommit=${GIT_COMMIT}' -X 'github.com/kanopy-platform/buildah-plugin/internal/version.pluginType=${PLUGIN_TYPE}'" -o /go/bin/app ./cmd/
+RUN go install github.com/awslabs/amazon-ecr-credential-helper/ecr-login/cli/docker-credential-ecr-login@latest
 
 FROM quay.io/buildah/stable:v1.31.0
 RUN groupadd -r app && useradd --no-log-init -r -g app app
+# Add ranges needed for buildah commands
+RUN echo app:10000:65536 >> /etc/subuid
+RUN echo app:10000:65536 >> /etc/subgid
```

Comment:
This is needed because I ran into [this](https://github.com/containers/podman/issues/1182#issuecomment-408844521)

Reply:
Do calls to buildah manifest need setuid and setgid? This isn't going to work under PSP or the privileged PodSecurity level.

My understanding was buildah manifest just took two existing OCI image definitions and added metadata .

Reply:
You're right, `buildah manifest` will spew warnings but not error if subuid and subgid ranges don't exist for user.

<details>
  <summary>output</summary>
```
time="2023-07-26T15:06:12Z" level=info msg="running: buildah manifest create public.ecr.aws/kanopy/buildah-plugin:multiarchtest2"
time="2023-07-26T15:06:12Z" level=warning msg="Reading allowed ID mappings: reading subuid mappings for user \"app\" and subgid mappings for group \"app\": no subuid ranges found for user \"app\" in /etc/subuid"
time="2023-07-26T15:06:12Z" level=warning msg="Found no UID ranges set aside for user \"app\" in /etc/subuid."
time="2023-07-26T15:06:12Z" level=warning msg="Found no GID ranges set aside for user \"app\" in /etc/subgid."
34107b840863fe03fe1cfd56cd045542cf9dfc856d163249c14528a2e28620e1
time="2023-07-26T15:06:12Z" level=info msg="running: buildah manifest add public.ecr.aws/kanopy/buildah-plugin:multiarchtest2 public.ecr.aws/kanopy/buildah-plugin:git-3afa39c-arm64"
time="2023-07-26T15:06:12Z" level=warning msg="Reading allowed ID mappings: reading subuid mappings for user \"app\" and subgid mappings for group \"app\": no subuid ranges found for user \"app\" in /etc/subuid"
time="2023-07-26T15:06:12Z" level=warning msg="Found no UID ranges set aside for user \"app\" in /etc/subuid."
time="2023-07-26T15:06:12Z" level=warning msg="Found no GID ranges set aside for user \"app\" in /etc/subgid."
34107b840863fe03fe1cfd56cd045542cf9dfc856d163249c14528a2e28620e1: sha256:1aceae525ba00988c461de81b4c8ff70d3ed0b1af3abf78530deef82a7fa6a40
time="2023-07-26T15:06:14Z" level=info msg="running: buildah manifest add public.ecr.aws/kanopy/buildah-plugin:multiarchtest2 public.ecr.aws/kanopy/buildah-plugin:git-3afa39c-amd64"
time="2023-07-26T15:06:14Z" level=warning msg="Reading allowed ID mappings: reading subuid mappings for user \"app\" and subgid mappings for group \"app\": no subuid ranges found for user \"app\" in /etc/subuid"
time="2023-07-26T15:06:14Z" level=warning msg="Found no UID ranges set aside for user \"app\" in /etc/subuid."
time="2023-07-26T15:06:14Z" level=warning msg="Found no GID ranges set aside for user \"app\" in /etc/subgid."
34107b840863fe03fe1cfd56cd045542cf9dfc856d163249c14528a2e28620e1: sha256:2096076ceb3f9d5f80db6e68b2f9c23b70007d3ef614427bb4438f302fadfda6
time="2023-07-26T15:06:16Z" level=info msg="running: buildah manifest push --all public.ecr.aws/kanopy/buildah-plugin:multiarchtest2"
time="2023-07-26T15:06:16Z" level=warning msg="Reading allowed ID mappings: reading subuid mappings for user \"app\" and subgid mappings for group \"app\": no subuid ranges found for user \"app\" in /etc/subuid"
time="2023-07-26T15:06:16Z" level=warning msg="Found no UID ranges set aside for user \"app\" in /etc/subuid."
time="2023-07-26T15:06:16Z" level=warning msg="Found no GID ranges set aside for user \"app\" in /etc/subgid."
Getting image list signatures
Copying 2 of 2 images in list
Copying image sha256:1aceae525ba00988c461de81b4c8ff70d3ed0b1af3abf78530deef82a7fa6a40 (1/2)
Getting image source signatures
Copying blob sha256:bcdf0bf7c1b3f5aba6597b1a632265b672f7aee5d442274db560bbf7f9f39d7a
Copying blob sha256:30bfbe1cefebb80f6f9d5d90c39fcdf9d6367bbc76fa68e33ea3b4391c0f9a77
Copying blob sha256:416f2232a731c0ecd4aa55b9592f0e749bf6e163e9bc584afa263dd020d69a12
Copying config sha256:2d06e55549ba2652b0e4964637172d4d56c1b26070c8a00bacd4c99d4a8b8df2
Writing manifest to image destination
Copying image sha256:2096076ceb3f9d5f80db6e68b2f9c23b70007d3ef614427bb4438f302fadfda6 (2/2)
Getting image source signatures
Copying blob sha256:abd8322911bb110a83c6084a6523249e7f54184ab64aa3cf457d7e93a54b4ea4
Copying blob sha256:90ac1ecaf92c8ae0cb37d29d8cc01b5802612c12edb933c369ad4c586ea94c6c
Copying blob sha256:7625c9a5501d49f8965b5ac95175a8a862bae39c3e3f5eb59604cb3f70157a6e
Copying config sha256:ac76caa89504d5d25e8e7c53cce19072783419569005dce7ffbc7b0e9b00a425
Writing manifest to image destination
Writing manifest list to image destination
Storing list signatures
```
</details>

I changed manifest.go to add in the manifest commands so you can try it out too.

> This isn't going to work under PSP or the privileged PodSecurity level.
Do they allow the `usermod` command?
If not, can we run container as root instead of app?
It doesn't seem to matter for manifest other than warnings, but just thinking of any future buildah commands